packages/block-library/src/latest-posts/edit.js

```diff
@@ -245,14 +337,28 @@ class LatestPostsEdit extends Component {
 
 export default withSelect( ( select, props ) => {
 	const { postsToShow, order, orderBy, categories } = props.attributes;
-	const { getEntityRecords } = select( 'core' );
+	const { getEntityRecords, getMedia } = select( 'core' );
+	const { getSettings } = select( 'core/block-editor' );
+	const { imageSizes } = getSettings();
 	const latestPostsQuery = pickBy( {
 		categories,
 		order,
 		orderby: orderBy,
 		per_page: postsToShow,
 	}, ( value ) => ! isUndefined( value ) );
 	return {
+		imageSizeOptions: map( imageSizes, ( { name, slug } ) => ( { value: slug, label: name } ) ),
+		getFeaturedMediaSourceUrl( featuredImageId, sizeSlug ) {
```

Comment:
Both the ID and the size slug are part of the component's props, so this could be implemented without introducing indirection through another function call. I.e., pass down the `featuredMediaSourceUrl`.

Reply:
Just to check, are you suggesting to augment [the `latestPosts` array](https://github.com/WordPress/gutenberg/blob/2f8bb9a3f917956dc752a1ca0f662976a5870c6b/packages/block-library/src/latest-posts/edit.js#L362) with the `featuredMediaSourceUrl` for each post? We can't just directly return `featuredMediaSourceUrl` since it's actually fetched per post.

Reply:
Yes